app/plugin_install.go

```diff
@@ -195,35 +195,38 @@ func (ch *Channels) InstallMarketplacePlugin(request *model.InstallMarketplacePl
 	if *ch.cfgSvc.Config().PluginSettings.EnableRemoteMarketplace {
 		var plugin *model.BaseMarketplacePlugin
 		plugin, appErr = ch.getRemoteMarketplacePlugin(request.Id, request.Version)
-		if appErr != nil {
+		// The plugin might only be prepacked and not on the Marketplace.
+		if appErr != nil && appErr.Id != "app.plugin.marketplace_plugins.not_found.app_error" {
 			return nil, appErr
 		}
```

Comment:
Is the following a potential failure scenario not covered by this change?
* pre-packaged plugin exists
* remote marketplace enabled, but fails to load, triggering an error other than "not found" (e.g. port blocked by firewall)
* installation of pre-packaged plugin still fails

Should we just log a warning here and never return an error from this block? We already return an error below if we fail to find a plugin altogether.

Reply:
That is a good suggestion :+1: 

Updated the code in c7b99c8f7